tests/conftest.py

```diff
@@ -221,7 +222,21 @@ def application_process(
 
 @fixture(scope="session")
 def docker_client() -> DockerClient:
-    return docker.from_env()
+    tests_id = secrets.token_hex(5)
+    docker_client = docker.from_env()
+    setattr(docker_client, "_gprofiler_test_id", tests_id)
+    yield docker_client
+
+    exited_ids: List[str] = []
+    exited_container_list = docker_client.containers.list(filters={"status": "exited", "label": tests_id})
+    for container in exited_container_list:
+        exited_ids.append(container.id)
+
+    pruned_ids = docker_client.containers.prune(filters={"label": tests_id}).get("ContainersDeleted", [])
+
+    for exited_id in exited_ids:
+        if exited_id not in pruned_ids:
+            raise Exception(f"Container with id {exited_id} has not been properly pruned")
```

Comment:
If we have multiple "non pruned" containers, we only get an exception for one. Lets create the full set of elements that appear in exited but not in pruned, and if that's a list longer than 0, raise an exception with them?